network/p2p/streams.go

```diff
@@ -160,6 +174,12 @@ func (n *streamManager) Disconnected(net network.Network, conn network.Conn) {
 		stream.Close()
 		delete(n.streams, conn.RemotePeer())
 	}
+
+	stream, ok = n.streams[conn.LocalPeer()]
```

Comment:
why is both local and remote needed here?

Reply:
remote tracking is the original code but http handler users local addr to store connection info in context so had to also add local